internal/util/update/update_test.go

```diff
@@ -678,6 +680,137 @@ func TestCommand_Run_failInvalidPath(t *testing.T) {
 	}
 }
 
+func TestCommand_Run_badUpstreamLockRepo(t *testing.T) {
+	repos, w, clean := testutil.SetupReposAndWorkspace(t, map[string][]testutil.Content{
+		testutil.Upstream: {
+			{
+				Pkg: pkgbuilder.NewRootPkg().
+					WithResource(pkgbuilder.DeploymentResource),
+				Branch: mainBranch,
+			},
+		},
+	})
+	defer clean()
+
+	w.PackageDir = testPackageName
+	dir := "/"
+	kf := kptfileutil.DefaultKptfile(testPackageName)
+	kf.Upstream = &kptfilev1alpha2.Upstream{
+		Type: kptfilev1alpha2.GitOrigin,
+		Git: &kptfilev1alpha2.Git{
+			Repo:      repos[testutil.Upstream].RepoDirectory,
+			Directory: dir,
+			Ref:       mainBranch,
+		},
+		UpdateStrategy: kptfilev1alpha2.ResourceMerge,
+	}
+	kf.UpstreamLock = &kptfilev1alpha2.UpstreamLock{
+		Type: kptfilev1alpha2.GitOrigin,
+		GitLock: &kptfilev1alpha2.GitLock{
+			Repo:      "fake",
+			Directory: dir,
+			Ref:       mainBranch,
+		},
+	}
+	testutil.AddKptfileToWorkspace(t, w, kf)
+
+	// Update the local package
+	err := Command{
+		Pkg: pkgtest.CreatePkgOrFail(t, w.FullPackagePath()),
+	}.Run()
+
+	if assert.Error(t, err) {
+		assert.Contains(t, err.Error(), "does not appear to be a git repository")
+	}
+}
+
+func TestCommand_Run_badUpstreamLockRef(t *testing.T) {
+	repos, w, clean := testutil.SetupReposAndWorkspace(t, map[string][]testutil.Content{
```

Comment:
For this situation I think we want to use the TestSetupManager so we actually get a fork in the workspace and then try to do the update with an incorrect ref. The way it is set up now, the repo is actually in a weird state since the `UpstreamLock` says the package has been fetched, but the resources doesn't actually exist on disk.

Reply:
Updated to use `TestSetupManager`